### PR TITLE
Added ability to use custom loggers for Flash and Flasher objects.

### PR DESCRIPTION
**Pull request by jonikula**
Added optional custom logger for Flash and Flasher objects. If no logger provided, the previous default ones are used.

Tests included.

### CODE REVIEW

**Comment by jonikula** [comment]
Hm, for some reason this works badly on python 3. Will investigate.

**Comment by YifeiZuo01** [comment]
feels like a bug there for decode and encode, feels like not every byte like 0xf2 can be decoded.

**Comment by jonikula** [comment]
Looks like most of those error messages were related to some destructor printing a ResourceError in py3 and not in py2. I got rid of those. Now looking into replicating the actual fail there.

**Comment by jonikula** [comment]
Hmpf, unable to reproduce the issue on my local VM with python 3.5. Which python version are we using in CI for python 3?

**Comment by jonikula** [comment]
@YifeiZuo01 you able to reproduce the issue locally?

**Comment by YifeiZuo01** [comment]
nope, I cannot, mine is also 3.5. Maybe ssh to the slave would help.
`62.44.193.186:3008`

**Comment by jonikula** [comment]
Yes, I managed to reproduce the issue. Fixed the test.

Basically, the test that fails flashes a binary and does not call reset. Next we read from the serial and assert that the command was not received (the application is not running yet). The read will return some carbagey data apparently and the decode called in python 3 version of the method will raise an exception, causing the test to return an error. This is not a problem on python 2 since we don't need to call the decode, so no exceptions are raised.

**Comment by jonikula** [comment]
@YifeiZuo01 @juhhov @jupe @VeliMattiLahtela reviews?

File: mbed_flasher/erase.py [hunk 1]
```diff
@@ -111,9 +111,11 @@ def wait_to_disappear(self, mount_point):
             if platform.system() == 'Windows':
                 proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
                 err = proc.stderr.read()
+                proc.communicate()
             else:
                 proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)
                 err = proc.stderr.read()
+                proc.communicate()
```
Comment (juhhov) [inline mbed_flasher/erase.py]: Could remove duplicate code
```
    if platform.system() == 'Windows':
        proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
    else:
        proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)

    err = proc.stderr.read()
    proc.communicate()
```

File: mbed_flasher/erase.py [hunk 2]
```diff
@@ -134,9 +136,11 @@ def runner(self, mount_point, filename):
             if platform.system() == 'Windows':
                 proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
                 out = proc.stdout.read()
+                proc.communicate()
             else:
                 proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)
                 out = proc.stdout.read()
+                proc.communicate()
```
Comment (juhhov) [inline mbed_flasher/erase.py]: Could remove duplicate code
```
    if platform.system() == 'Windows':
        proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
    else:
        proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)

    err = proc.stderr.read()
    proc.communicate()
```

**Pull request merged**